### PR TITLE
Update dockerfile and makefile

### PR DESCRIPTION
**Pull request by oelbert**
**Description**
This PR adds a dockerfile to ease development and testing of pyFV3, similar to fv3core, and updates the makefile to streamline everything and shed some of the AI2 artifacts. This also adds an example jupyter notebook that uses the NDSL infrastructure to run a pyFV3 stencil on the cubed-sphere and run halo updates, and fixes the doubly-periodic test.

 - `Dockerfile`: Added file, contains recipe for an image that can run tests, run pyFV3 code, and visualize model outputs
 - `Makefile`: Replaced fvcore makefile with one that doesn't reference vcm/ai2cm, but successfully gets test data, builds a docker image, runs tests, and launches an ipynb server from the image
 - `examples/notebook/test_functionality.ipynb`: Jupyter notebook that uses most of the NDSL model functionality to run a stencil in parallel
 - `tests/mpi/test_doubly_periodic.py`: Fixed metric terms setup so it reads an eta file that should live in test data

**How Has This Been Tested?**
To test this I ran all of the savepoint tests and the doubly periodic test using the makefile, and used `test_functionality.ipynb` to ensure that the pyFV3 and NDSL code is available and functioning correctly inside the container.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] New check tests, if applicable, are included
- [ ] Targeted model if this changed was triggered by a model need/shortcoming

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,96 +1,106 @@
...
-PYTEST_PARALLEL=$(MPIRUN_CALL) python3 -m mpi4py -m pytest --maxfail=1 --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS) $(THRESH_ARGS) -m parallel $(FV3_PATH)/tests/savepoint
-ifeq ($(DEV),y)
-	VOLUMES += -v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $(FV3UTIL_DIR):/external/util -v $(CWD)/external/dsl:/external/dsl -v $(CWD)/external/stencils:/external/stencils
+ifeq ($(DEV), y)
+	VOLUMES += -v $(CWD):/pyFV3
 endif
-CONTAINER_CMD?=$(CONTAINER_ENGINE) run $(RUN_FLAGS) $(VOLUMES) $(PACE_IMAGE)
 
-clean:
+TEST_DATA_TARFILE = $(TEST_DATA_VERSION)_$(TEST_RESOLUTION)_$(NUM_RANKS)_ranks_$(TEST_TYPE).tar.gz
```
Comment (FlorianDeconinck) [inline Makefile]: Yeah that's nasty, my bad. I should clean naming before we merge.

**Review by FlorianDeconinck — COMMENTED** [review]
LGTM

I am going to correct the data naming on the datashare.

Question: how we do we test the notebook? It be good to have it in `ci` so it doesn't get broken

**Comment by oelbert** [comment]
Good question about the notebook. I think nbconvert should cover it if the CI platform can support 6 ranks and has ipyparallel?

File: Makefile (hunk 1 above)
Comment (oelbert) [reply Makefile]: thank you 🙏

Comment (FlorianDeconinck) [reply Makefile]: Done

**Comment by FlorianDeconinck** [comment]
> Good question about the notebook. I think nbconvert should cover it if the CI platform can support 6 ranks and has ipyparallel?

I very much expect it, yes.

**Comment by FlorianDeconinck** [comment]
Please merge in `develop` instead of `main`

**Review by FlorianDeconinck — DISMISSED** [review]
Logged the notebook to `ci` task and it's good for me

**Review by FlorianDeconinck — APPROVED** [review]

**Review by fmalatino — APPROVED** [review]
lgtm

**Pull request merged**